testing-unit-common/src/main/java/com/bumble/appyx/testing/unit/common/util/InteropBuilderStub.kt

```diff
@@ -0,0 +1,46 @@
+package com.bumble.appyx.testing.unit.common.util
+
+import androidx.lifecycle.Lifecycle
+import com.bumble.appyx.core.builder.Builder
+import com.bumble.appyx.core.modality.BuildContext
+import com.bumble.appyx.core.node.Node
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class InteropBuilderStub<P>(
+    val delegate: (BuildContext, P) -> Node = { _, _ -> NodeStub() },
+) : Builder<P>() {
+
+    private val _createdNodes = ArrayList<Node>()
+
+    val last: Node?
+        get() = _createdNodes.lastOrNull()
+
+    var lastParam: P? = null
+        private set
+
+    override fun build(buildContext: BuildContext, payload: P): Node =
+        delegate(buildContext, payload).also {
+            _createdNodes += it
+            lastParam = payload
+        }
+
+    fun assertLastParam(param: P) {
+        assertEquals(param, lastParam)
+    }
+
+    fun assertLastParam(assert: P.() -> Boolean) {
+        assertTrue(
+            lastParam != null && assert(lastParam!!),
+            "$lastParam does not satisfy requirements"
+        )
+    }
+
+    fun assertCreatedNode() {
+        assertTrue(last != null, "Has not created any node")
+    }
+
+    fun assertLastNodeState(state: Lifecycle.State) {
+        assertEquals(state, last?.lifecycle?.currentState)
```

Comment:
for now if last?.lifecycle?.currentState returns null, you know that the last node was null. however potentially the API could change in the future which may make this test have a false positive/negative (i.e. how lifecycle is nullable, but the last node was not null, making it confusing to tell why the test failed, or worse it passes).

I proposed the following (because we don't seem to have soft asserts):

assertCreatedNode()
assertEquals(state, last!!.lifecycle.currentState)
This way if the API of lifecycle changes to be nullable, the code would not compile.